content/docs/schema-advanced.md

```diff
@@ -0,0 +1,318 @@
+                  type: "string",
+                  ui: {
+                    component: "textarea",
+                  },
+                },
+              ],
+            },
+          ],
+        },
+      ],
+    },
+  ],
+});
+```
+​
+## Query
+​
+To retrieve a single `recipe`, you would use a query like this one.
+​
+Notice that, for `sections`, we are using `__typename` to deobfuscate the available `templates`. While `__typename` can be omitted in the query, keeping it allows for comparisons later where you might want to use a different render for each `template` type.
+​
+Because `meta` does not have any `templates`, you do not need to deobfuscate and can access its properties directly.
+​
+```ts, copy
+// pages/recipe/[filename].tsx
+import { getStaticPropsForTina, gql } from "tinacms";
+​
+export const getStaticProps = async ({ params }) => {
+  const recipe = await getStaticPropsForTina({
+    query: gql`
+      query GetRecipeDocument($relativePath: String!) {
+        getRecipeDocument(relativePath: $relativePath) {
+          data {
+            datePublished
+            name
+            photo
+            description
+            meta {
+              prep
+              cook
+              servings
+            }
+            sections {
+              __typename
+              ... on RecipeSectionsIngredients {
+                items
+                details
+              }
+              ... on RecipeSectionsDirections {
+                steps
+                details
+              }
+              ... on RecipeSectionsNutrition {
+                details
+              }
+            }
+          }
+        }
+      }
+    `,
+    variables: { relativePath: `${params.filename}.md` },
+  });
+​
+  return {
+    props: {
+      ...recipe,
+    },
+  };
+};
+```
+​
+This query will return an `Object` in the shape of:
+​
+```ts, copy
+{
+  "data": {
+    "getRecipeDocument": {
+      "data": {
+        "datePublished": string,
+        "name": string,
+        "photo": string,
+        "description": string,
+        "meta": {
+          "prep": string,
+          "cook": string,
+          "servings": string,
+        },
+        "sections": [
+          {
+            "__typename": "RecipeSectionsIngredients",
+            "items": [string],
+            "details": string,
+          },
+          {
+            "__typename": "RecipeSectionsDirections",
+            "steps": [string],
+            "details": string,
+          },
+          {
+            "__typename": "RecipeSectionsNutrition",
+            "details": string,
+          }
+        ]
+      }
+    }
+  }
+}
+```
+​
+## Rendering
+​
+The important step here is correctly retrieve the queried `data` out of `props`. The common pattern is `data: { query: { data: document }}`.
+​
+Notice that `sections` is iterated via `map()` utilizing the `__typename` to determine how each `section` should be rendered. You could also use `_template`.
```

Comment:
```suggestion
Notice that `sections` is iterated via `map()` utilizing the `__typename` to determine how each `section` should be rendered.
```
`_template` isn't supported here, despite some misleading docs elsewhere (which should also be fixed)

Reply:
Thanks for catching this @jeffsee55 , I added that before I realized we weren't doing it.